### ISSUE TITLE
passing parameter value from parent node

### ISSUE THREAD

**Issue by pli2014** [graphql-java/graphql-java#111]
for Example: 
```graphql
queryMyFriends(userId:xxxx){
        friendId:id
        name
        profile(id:$friendId){
             level
             score
        }
}
```
Got my friends list data and corresponding profile info,  which need to be passing friendId to this profile node. 
In DataFetcher interface, we had to fetch this friendId variable from parent data source.  do you implement this feature?   context variable could be passed in parent-child calling situation.

**Comment by kaqqao** [comment]
I believe you shouldn't even need the friendID in this situation... In your profile DataFetcher, you have access to the entire source object via DataFetchingEnvironment#getSource (which would be the friend in your case), so you don't specifically need to pass the ID.

**Comment by pli2014** [comment]
if queryMyFriends function and profile are independent, I think it is necessary to implement this feature. recently, I completed this feature, so I raised this issue to you guys.  
In my opinion, I mapped function calling to method of Java based code in my product, which brought many benefits for me, my target is implement Multidimensional Data Aggregation.  In my product, different data service need to be aggregated by Graphql. So I encountered this context-based passing variable.
